Scripts/ExtractDomainFromUrlFormat/ExtractDomainFromUrlFormat.py

```diff
@@ -62,26 +64,43 @@ def extract_domain(the_input):
         # Not ATP Link or Proofpoint URL so just unescape
         else:
             the_input = unescape_url(the_input)
-        is_url = domain = get_tld(the_input, fail_silently=True)
+
+        is_url = domain = get_tld(the_input, fail_silently=True, as_object=True)
+
+        # handle fqdn if needed
+        if isFQDNextract and domain:
+            # get the subdomain using tld.subdomain
+            subdomain = domain.subdomain
+            if (subdomain):
+                fqdn = "{}.{}".format(subdomain, str(domain))
 
     # Extract domain itself from a potential subdomain
     if domain_from_mail or not is_url:
         full_domain = 'https://'
         full_domain += domain_from_mail if domain_from_mail else the_input
         # get_tld fails to parse subdomain since it is not URL, over-ride error by injecting protocol.
-        domain = get_tld(full_domain, fail_silently=True)
+        domain = get_tld(full_domain, fail_silently=True, as_object=True)
+
+        # handle fqdn if needed
+        if isFQDNextract and domain:
+            subdomain = domain.subdomain
+            if (subdomain):
+                fqdn = "{}.{}".format(subdomain, str(domain))
 
     # convert None to empty string if needed
-    domain = '' if not domain else domain
-    return domain
+    result = domain if not isFQDNextract else fqdn
+    result = '' if not result else result
+    return result
 
 
-domains = []
+results = []
 the_input = demisto.args().get('input')
+isFQDNextract = demisto.args().get('extractFQDN')
```

Comment:
for input `false` (string) it will still extract FQDN.
this could happen when running from a playbook.

Reply:
Actually it will always by string (script argument has no type) but I fixed this in the next commit